File: ContentProDemo/app/src/main/java/edu/cs4730/contentprodemo/data/Square.java
package edu.cs4730.contentprodemo.data;

import android.database.Cursor;
import android.database.MatrixCursor;

/**
 * * A very simple data class that return a cursors with the square of the values.
 */
public class Square {

    String[]  Column;

    public Square() {
      Column = new String[] { "number", "square"};
    }

    //getone returns the val and the square of the val in cursor.
    public Cursor getone(int val) {
        //must return a Cursor, MatrixCursor is an editable cursor.
        MatrixCursor myCursor = new MatrixCursor(Column);
        myCursor.addRow(new Object[] { val, val*val});
        return myCursor;
    }

    //returns the number and number of the number from 1 to 10.
    public Cursor getall() {
        MatrixCursor myCursor = new MatrixCursor(Column);
        for (int i=1; i<11; i++) {
            myCursor.addRow(new Object[] { i, i*i});
        }
        return myCursor;
    }
}

File: ContentProDemo_kt/app/src/main/java/edu/cs4730/contentprodemo_kt/data/Square.kt
package edu.cs4730.contentprodemo_kt.data

import android.database.Cursor
import android.database.MatrixCursor

/**
 * * A very simple data class that return a cursors with the square of the values.
 */
class Square {
    private var Column: Array<String> = arrayOf("number", "square")

    //getone returns the val and the square of the val in cursor.
    fun getone(value: Int): Cursor {
        //must return a Cursor, MatrixCursor is an editable cursor.
        val myCursor = MatrixCursor(Column)
        myCursor.addRow(arrayOf<Any>(value, value * value))
        return myCursor
    }

    //returns the number and number of the number from 1 to 10.
    fun getall(): Cursor {
        val myCursor = MatrixCursor(Column)
        for (i in 1..10) {
            myCursor.addRow(arrayOf<Any>(i, i * i))
        }
        return myCursor
    }
}
